From a43dcdf8271b816c3f7b692aafd24572ccf2b2cf Mon Sep 17 00:00:00 2001
From: Zubair Lutfullah Kakakhel <Zubair.Kakakhel@imgtec.com>
Date: Wed, 19 Jul 2017 15:27:20 +0100
Subject: [PATCH] HACK: MIPS: Disable R4K_SYNC

Signed-off-by: Zubair Lutfullah Kakakhel <Zubair.Kakakhel@imgtec.com>
---
 arch/mips/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/mips/Kconfig b/arch/mips/Kconfig
index b3c5bde..0a06c95 100644
--- a/arch/mips/Kconfig
+++ b/arch/mips/Kconfig
@@ -2365,7 +2365,7 @@ config MIPS_CPS
 	select MIPS_CPC
 	select MIPS_CPS_PM if HOTPLUG_CPU
 	select SMP
-	select SYNC_R4K if (CEVT_R4K || CSRC_R4K)
+#	select SYNC_R4K if (CEVT_R4K || CSRC_R4K)
 	select SYS_SUPPORTS_HOTPLUG_CPU
 	select SYS_SUPPORTS_SMP
 	select WEAK_ORDERING
-- 
2.10.2

